src/asMiddleware.js

```diff
@@ -1,19 +1,30 @@
 'use strict'
 
 module.exports = function asMiddleware (engine) {
-  return function engineAsMiddleware (req, res, next, end) {
```

Comment:
This does change the return type of this function, from `undefined` to `Promise`. Is this change required to remove `async`? I don't immediately see the connection.

Reply:
The reason for the change is in order to `await engine._runMiddlewareDown` on the subsequent line, but I believe it is convertible to `.then`/`.catch`. I'll give that a shot.

Reply:
f919617